src/dragdrop/ranking-select-to-rank.ts

```diff
@@ -0,0 +1,163 @@
+
+    if (this.isDraggedElementUnranked && this.isDropTargetRanked) {
+      fromIndex = unRankingChoices.indexOf(this.draggedElement);
+      if (rankingChoices.length === 0) {
+        toIndex = 0;
+      } else {
+        toIndex = rankingChoices.indexOf(this.dropTarget);
+      }
+      this.selectToRankEnabled(questionModel, fromIndex, toIndex);
+      this.doUIEffects(dropTargetNode, fromIndex, toIndex);
+      return;
+    }
+
+    if (this.isDraggedElementRanked && this.isDropTargetRanked) {
+      fromIndex = rankingChoices.indexOf(this.draggedElement);
+      toIndex = rankingChoices.indexOf(this.dropTarget);
+      this.reorderRankedItem(questionModel, fromIndex, toIndex);
+      this.doUIEffects(dropTargetNode, fromIndex, toIndex);
+      return;
+    }
+
+    if (this.isDraggedElementRanked && !this.isDropTargetRanked) {
+      fromIndex = rankingChoices.indexOf(this.draggedElement);
+      if (unRankingChoices.length === 0) {
+        toIndex = 0;
+      } else {
+        toIndex = unRankingChoices.indexOf(this.dropTarget);
+      }
+      this.unselectFromRank(questionModel, fromIndex);
+      this.doUIEffects(dropTargetNode, fromIndex, toIndex);
+      return;
+    }
+  }
+
+  private doUIEffects(dropTargetNode: HTMLElement, fromIndex: number, toIndex:number) {
+    const questionModel: any = this.parentElement;
+    const isDropToEmptyRankedContainer = this.dropTarget === "to-container" && questionModel.isEmpty();
+    const isNeedToShowIndexAtShortcut = !this.isDropTargetUnranked || isDropToEmptyRankedContainer;
+    const shortcutIndex = isNeedToShowIndexAtShortcut ? toIndex + 1 : null;
+
+    this.updateDraggedElementShortcut(shortcutIndex);
+
+    if (fromIndex !== toIndex) {
+      dropTargetNode.classList.remove("sv-dragdrop-moveup");
+      dropTargetNode.classList.remove("sv-dragdrop-movedown");
+      questionModel.dropTargetNodeMove = null;
+    }
+
+    if (fromIndex > toIndex) {
+      questionModel.dropTargetNodeMove = "down";
+    }
+
+    if (fromIndex < toIndex) {
+      questionModel.dropTargetNodeMove = "up";
+    }
+  }
+
+  private get isDraggedElementRanked() {
+    return this.parentElement.rankingChoices.indexOf(this.draggedElement) !== -1;
+  }
+
+  private get isDropTargetRanked() {
+    if (this.dropTarget === "to-container") return true;
+    return this.parentElement.rankingChoices.indexOf(this.dropTarget) !== -1;
+  }
+
+  private get isDraggedElementUnranked() {
+    return !this.isDraggedElementRanked;
+  }
+
+  private get isDropTargetUnranked() {
+    return !this.isDropTargetRanked;
+  }
+
+  public selectToRankEnabled(questionModel: QuestionRankingModel, fromIndex: number, toIndex: number): void {
+    const rankingChoices = questionModel.rankingChoices;
+    const unRankingChoices = questionModel.unRankingChoices;
+    const item = unRankingChoices[fromIndex];
+
+    rankingChoices.splice(toIndex, 0, item);
+    questionModel.setPropertyValue("rankingChoices", rankingChoices);
```

Comment:
After fix the code duplication the last line and probably some others can be moved to another function and do not used in all 3 functions.